lint_test/avoid_unused_parameters_test.dart

```diff
@@ -0,0 +1,109 @@
+// ignore_for_file: prefer_const_declarations
+// ignore_for_file: unnecessary_nullable_for_final_variable_declarations
+// ignore_for_file: unused_local_variable
+// ignore_for_file: unused_element
+// ignore_for_file: newline-before-return
+
+import 'package:flutter/material.dart';
+
+/// Check the `avoid-unused-parameters` rule
+
+// expect_lint: avoid-unused-parameters
+void fun(String s) {
+  return;
+}
+
+class TestClass {
+  // expect_lint: avoid-unused-parameters
+  static void staticMethod(int a) {}
+
+  // expect_lint: avoid-unused-parameters
+  void method(String s) {
+    return;
+  }
+
+  void methodWithUnderscores(int _) {}
+}
+
+class TestClass2 {
+  void method(String _) {
+    return;
+  }
+}
+
+class SomeOtherClass {
+  // expect_lint: avoid-unused-parameters
+  void method(String s) {
+    return;
+  }
+}
+
+class SomeAnotherClass extends SomeOtherClass {
+  @override
+  void method(String s) {}
+}
+
+void someOtherFunction(String s) {
+  print(s);
+  return;
+}
+
+class SomeOtherAnotherClass {
+  void method(String s) {
+    print(s);
+    return;
+  }
+
+  // expect_lint: avoid-unused-parameters
+  void anonymousCallback(Function(int a) cb) {}
```

Comment:
```suggestion
  // expect_lint: avoid-unused-parameters
  void anonymousCallback(Function(int a) cb) {}
  void callbackUser() {
     // expect_lint: avoid-unused-parameters
    anonymousCallback((int a) {});
  }
```

Reply:
Same issue as with [this comment](https://github.com/solid-software/solid_lints/pull/49#pullrequestreview-1628707372)